client/vm.go

```diff
@@ -292,20 +302,34 @@ func (c *Client) GetVm(vmReq Vm) (*Vm, error) {
 	return &vms[0], nil
 }
 
-func (c *Client) GetVms() ([]Vm, error) {
-
+func (c *Client) GetVms(vm Vm) ([]Vm, error) {
+	var vms []Vm
 	var response map[string]Vm
-	err := c.GetAllObjectsOfType(Vm{PowerState: "Running"}, &response)
+	if vm.Container == "" && vm.PowerState == "" {
```

Comment:
Rather than introduce this conditional branching, I would prefer if we updated the previous callers to pass.

```
# Previous usage
c.GetVms()

# New usage
c.Get(Vm{PowerState: "Running"})
```